Problem037/C++/solution_1.cpp

```diff
@@ -0,0 +1,42 @@
+#include<bits/stdc++.h>
+using namespace std;
+
+const int much = 1000000;
+bool isPrime [much];
+
+void sieve (){
+	memset (isPrime, true, sizeof(isPrime));
+	isPrime[0] = isPrime [1] = false;
+	for (int i = 2; i < much; ++i)
+		for (int j = (i << 1); j < much; j += i)
+			isPrime [j] = false;
+}
+
+bool test (int t)
+{
+	int orig = t, cnt = 0, tmp10 = 1;
+	while (t > 0){
+		if (!isPrime [t])
```

Comment:
There's no need for parenthesis before the square brackets, please remove them.

Reply:
which parenthesis are you talking about here :p

Reply:
As I can see, I have a bunch of bad coding habits! :(

Reply:
Sorry about that, I expressed myself badly. I meant that there's no spaces before square brackets and that should be written like this:

`if (!is_prime[t]) { ...`

Take a not that on C++ we prefer to use _snake_case_ instead of _camelCase_.

Reply:
There is also a space needed after the parenthesis:

`while (t > 0) { ...`